src/uwtools/rocoto.py

```diff
@@ -134,20 +133,14 @@ def validate_rocoto_xml(input_xml: OptionalPath) -> bool:
     :param input_xml: Path to rendered XML file.
     :return: Did the XML file conform to the schema?
     """
-
-    # Validate the XML.
+    with readable(input_xml) as f:
+        tree = etree.parse(f)
     with open(_rocoto_schema_xml(), "r", encoding="utf-8") as f:
         schema = etree.RelaxNG(etree.parse(f))
-    with readable(input_xml) as f:
-        xml = f.read()
-    tree = etree.fromstring(bytes(xml, encoding="utf-8"))
-    success = schema.validate(tree)
-
-    # Log validation errors.
-    errors = str(etree.RelaxNG.error_log).split("\n")
-    log_method = log.error if len(errors) else log.info
-    log_method("%s Rocoto validation error%s found", len(errors), "" if len(errors) == 1 else "s")
-    for line in errors:
-        log.error(line)
-
-    return success
+    valid = schema.validate(tree)
+    nerr = len(schema.error_log)
+    log_method = log.info if valid else log.error
+    log_method("%s Rocoto validation error%s found", nerr, "" if nerr == 1 else "s")
+    for err in list(schema.error_log):
```

Comment:
Is `list` needed here? `schema.error_log` is already an iterable, no?

Reply:
Nope, turns out its non-iterable too, annoyingly